cupy/random/_bit_generator.pyx

```diff
@@ -0,0 +1,184 @@
+import threading
+
+import numpy
+
+from libc.stdint cimport intptr_t, uint64_t, uint32_t
+
+import cupy
+from cupy.cuda import curand
+from cupy.cuda cimport stream
+from cupy.core.core cimport ndarray
+
+
+cdef extern from 'cupy_distributions.cuh' nogil:
+    cppclass curandState:
+        pass
+    cppclass curandStateMRG32k3a:
+        pass
+    cppclass curandStatePhilox4_32_10_t:
+        pass
+
+    cdef enum _RandGenerators '::RandGenerators':
+        CURAND_XOR_WOW
+        CURAND_MRG32k3a
+        CURAND_PHILOX_4x32_10
+
+    void init_curand_generator(int generator, intptr_t state_ptr, uint64_t seed, ssize_t size, intptr_t stream);
+    void interval_32(int generator, intptr_t state, intptr_t out, ssize_t size, intptr_t stream, int mx, int mask);
+    void interval_64(int generator, intptr_t state, intptr_t out, ssize_t size, intptr_t stream, uint64_t mx, uint64_t mask);
+    void beta(int generator, intptr_t state, intptr_t out, ssize_t size, intptr_t stream, double a, double b);
+    void exponential(int generator, intptr_t state, intptr_t out, ssize_t size, intptr_t stream);
+
+_UINT32_MAX = 0xffffffff
+_UINT64_MAX = 0xffffffffffffffff
+
+
+class BitGenerator:
```

Comment:
Maybe we can also hide this class (rename to `_BitGenerator`)?

Reply:
Looks like NumPy exposes this object, but given that it's a base class which is useless without subclassing, I wonder how useful it is...